04.ls/lib/ls_command.rb

```diff
@@ -9,8 +9,7 @@ def initialize(path)
   end
 
   def execute
-    files = Dir.entries(@path).sort.map { |entry| LsFile.new(entry) }
-    files.reject!(&:hidden?)
-    Formatter.format(files)
+    ls_files = Dir.entries(@path).sort.map { |entry| LsFile.new(entry) }.reject(&:hidden?)
```

Comment:
```suggestion
    ls_files = Dir
      .entries(@path)
      .sort
      .map { |entry| LsFile.new(entry) }
      .reject(&:hidden?)
```

メソッドチェーンを多用するときは改行した方が処理が読みやすかったりします。

ただ、ちょっと気になるのは、前回のコメントは「破壊的メソッドを使うのではなく、非破壊的メソッドを使った上でいい名前の新しい変数に代入しようね」というのが僕の意図でした。

これ以外も、  https://github.com/hashimoto-kei/ruby-practices/pull/5#discussion_r1953673130 や https://github.com/hashimoto-kei/ruby-practices/pull/5#discussion_r1952037337 も「いい名前を考えてね」という意図でコメントしたのですが、いずれも「変数を無くす」「メソッドを無くす」というアプローチで解決していたのが、ちょっと引っかかっていました。

プログラムがシンプルなあいだは「変数を無くす」「メソッドを無くす」でも可読性を維持できますが、この先どんどん仕様が複雑になっていくので、そうなると「変数を無くす」「メソッドを無くす」という逃げ道が使いにくくなっていきます。

というわけで、ls2以降はいい名前を考える作業からあまり逃げないでほしいな〜というのが、個人的な思いです。

まあ、命名って難しいんですけどね！！w
https://x.com/jnchito/status/1281188190905196544

Reply:
コメントありがとうございます！

変数やメソッドの命名は難しいですね。。

ls2 以降は逃げられなくなっていくと思うので、
どんな命名をしているかという観点で ls1 の模範解答や他の方の提出物を見てみようと思います。